### DISCUSSION TITLE
Database sync across multiple environments

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8100] [Ideas]
> Created by: Ben Parizek (ben@barrelstrengthdesign.com) on 2015/03/12 23:19:32 +0000
> Votes at time of UserVoice import: 230

-----------------------------------

Managing database changes across multiple environments is hard.

Make it easier!

This thread on Stack Exchange is a good discussion of different methods people are currently using:
http://craftcms.stackexchange.com/questions/10/whats-the-best-practice-for-handling-data-migration-and-organization-across-dev

**Comment by angrybrad** [comment]
> Posted by Andrew M Stoltz (astoltz@gmx.com) on 2015/03/22 01:35:42 +0000
-----------------------------------

The database creates GUIDs for all data which could be useful for tracking what records are the same across environments. However, this field is not exposed to models for reading or writing so selectively pushing changes isn&#39;t available through the core API. A change to core would be necessary.

Additionally, base model doesn&#39;t cast Booleans (https://github.com/pixelandtonic/Craft-Release/pull/2) so information from the database will be an integer, but the same data from a form submission might be &quot;&quot; or &quot;1&quot; (strings).

**Comment by angrybrad** [comment]
> Posted by Peter Tell (peter.tell@gmail.com) on 2015/03/09 16:26:45 +0000
-----------------------------------

That plugin is great Florian, but as I understand it, it takes a copy of the Staging DB and completely overwrites the Production DB.

This feature request is so crucial because it would be amazing to sync the data between two systems, so that you could truly stage content in a staging environment while production &quot;user submitted&quot; data was still preserved.

For example, if people were submitting contact forms while you were in the process of staging content, you wouldn&#39;t want all that information blown away after the staged content moves live.

**Comment by angrybrad** [comment]
> Posted by Florian (fw@demodern.de) on 2015/03/13 06:38:29 +0000
-----------------------------------

For now check this out:
https://github.com/70kft/craft-golive

**Comment by kruncher** [comment]
Is there any progress on this feature request / enhancement?

**Comment by angrybrad** [comment]
@kruncher Project Config is a step in this direction https://docs.craftcms.com/v3/project-config.html

**Comment by jan10** [comment]
@kruncher i found a plugin for that: https://unionco.github.io/craft-plugin-docs/craft-sync-db/#overview

Works perfect :)
